### DISCUSSION TITLE
Breakpoints set in installed python files get grayed out when running a jupyter cell

### DISCUSSION THREAD

**Discussion by rbavery** [microsoft/vscode-jupyter#13394] [Questions and Answers]
I'm trying to debug a python file that's installed by mambaforge by setting a breakpoint in a jupyter cell.

I feel like I'm doing all the right things, justmycode for jupyter is false in my launch.json

```
    "version": "0.2.0",
    "configurations": [
        {    "name": "jupyter",
            "jupyter.debugJustMyCode": false,
            "justMyCode": false,
        },
```

and I can set breakpoints in my own python module that is installed editable with pip

but when I run "Debug Cell" after setting the breakpoint in the file "/home/work/mambaforge/envs/slickformer/lib/python3.10/site-packages/transformers/feature_extraction_utils.py" the breakpoint gets grayed out. And it looks like the kernel is actually running a tmp file.

```
Exception has occurred: TypeError
is_floating_point(): argument 'input' (position 1) must be Tensor, not list
  File "/home/work/tmp/ipykernel_3298/812386295.py", line 1, in <module>
    trainer.fit(model, datamodule=onef_dm)
TypeError: is_floating_point(): argument 'input' (position 1) must be Tensor, not list
```

How are folks debugging code in system libraries from jupyter cells? Is there a configuration I'm missing? Or is this not possible?

Another potential wrinkle is that I'm running this in a DevContainer. But since I'm able to set breakpoints in my notebook file and editable python module, it feels like debugging installed python libraries is within reach?

After writing this I found this issue: https://github.com/microsoft/vscode-jupyter/issues/12826 which indicates maybe the folder path for my mambaforge installation and project directory needs to be the same? Is this still the case or has this been addressed?

**Comment by amunger** [comment]
@roblourens - is this expected to work?

**Comment by roblourens** [comment]
The launch.json is not used for jupyter debugging. `jupyter.debugJustMyCode` is a setting for your settings.json file
